src/main/java/com/minecolonies/coremod/colony/managers/RaidManager.java

```diff
@@ -461,23 +489,52 @@ private BlockPos findSpawnPointInDirections(final BlockPos center, final Directi
                 }
                 else
                 {
-                    if (WorldUtil.isBlockLoaded(colony.getWorld(), tempPos.offset(dir2, 16))
-                          && WorldUtil.isBlockLoaded(colony.getWorld(), tempPos.offset(dir2, 32))
-                          && WorldUtil.isBlockLoaded(colony.getWorld(), tempPos.offset(dir1, 16)))
+                    if (areChunksLoadedForRaidSpawn(tempPos, dir2.getA(), dir1.getB()))
                     {
-                        if (isValidSpawnPoint(tempPos.offset(dir2, 16)))
+                        if (isValidSpawnPoint(tempPos.offset(dir2.getA(), 16)))
                         {
-                            spawnPos = tempPos.offset(dir2, 16);
+                            spawnPos = tempPos.offset(dir2.getA(), 16);
                         }
-                        tempPos = tempPos.offset(dir2, 16);
+                        tempPos = tempPos.offset(dir2.getA(), 16);
                     }
                     else
                     {
                         break;
                     }
                 }
-            }
 
+                // Direction two
+                if (random.nextBoolean())
```

Comment:
I wonder if there is a good way to encapsulate this into a method, seems to be almost 4 times the same code

Reply:
wondered too, but difficult as its a conditioned assign to two different vars